File: stringLoop/stringLoop.go
package main

func CharCount(str string) map[string]int {
	result := make(map[string]int)

	for i := 0; i < len(str); i++ {
		char := string(str[i])
		//if the char is a number/letter AND is a key in the map, add one to count
		if result[char] > 0 {
			result[char]++
			//if the char is a number/letter AND not in the object, add it and set to 1
		} else {
			result[char] = 1
		}
	}

	return result
}

File: patterns/stringLoop.js
const charCount = (str) => {
// make object to return at the end
    let obj = {}
    // loop over string, for each char...
    for (let char of str) {
        char = char.toLowerCase()

        if (/[a-z0-9]/.test(char)) {
            obj[char] = ++obj[char] || 1;
        }
    }

    return obj
    //if the char is a number/letter AND is a key, add one to the object
    // if the char is a number/letter AND not in object, add it to object and set elem to 1
    // if char is something else (garbage code) don't do anything
    // return obj at end
}

export {charCount}